### PR TITLE
Add code block substitution and conditional quick start in doc getting-started

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request is an alternative approach to PR #4003. As suggested by @adonath it uses a new sphinx directive for substitution in code block and is applied for the quick start section. A conditional statement is added not to include the quick start for dev versions.

Note that here the substitution is added in the code base. Conversely we can rely on the external package: 
https://github.com/adamtheturtle/sphinx-substitution-extensions

**Dear reviewer**

### CODE REVIEW

File: docs/sphinx_utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
```
Comment (adonath) [inline docs/sphinx_utils.py:1]: I think it's fine to keep this on our side, however the other sphinx extensions we have are in `gammapy/utils/docs.py`, maybe move to there?

File: docs/getting-started/index.rst [hunk 2]
```diff
@@ -55,66 +55,9 @@ Installation
         :text: Learn more
         :classes: btn-secondary stretched-link
 
-.. _quickstart-setup:
+.. ifconfig:: not 'dev' in version
```
Comment (adonath) [inline docs/getting-started/index.rst:58]: Does this mean the the Quickstart is not included at all for dev versions? I guess this mostly makes sense, alternatively we could just work with a place-holder like:

```
https://gammapy.org/download/install/gammapy-x.x.x-environment.yml
```

**Review by adonath — DISMISSED** [review]
Thanks @registerrier, this looks like an elegant solution to me!

File: docs/sphinx_utils.py (hunk 1 above)
Comment (registerrier) [reply docs/sphinx_utils.py]: Done. 
I did not manage to use exactly the same logic as was previously used. So I added the `setup` in the `conf.py`.

Should we test this code? I suppose we should, but it is not clear how. and what to test.

File: docs/getting-started/index.rst (hunk 2 above)
Comment (adonath) [reply docs/getting-started/index.rst:58]: I guess the downside of hiding it for dev version is that we don't see the rendered output even when changes are being made? But I have no strong opinion there...

Comment (registerrier) [reply docs/getting-started/index.rst:58]: Yes Indeed. With the `ifconfig` scheme you can only test the result on a tagged version, so hiding the section completely is a bit annoying.

Comment (adonath) [reply docs/getting-started/index.rst]: +1 for the `X.Y.Z` I think this makes it obvious, that there is no env file for dev versions...

**Comment by adonath** [comment]
Two more things:
- Update dev docs?
- Still include the argument forwarding for the `Makefile` from #4003? It is used for the notebooks...

**Comment by registerrier** [comment]
I can open a new PR for the Makefile as it is a slightly different matter

**Comment by adonath** [comment]
Idea from dev meeting: add a warning box for dev builds saying there is no quickstart setup

**Review by adonath — APPROVED** [review]
Thanks @registerrier! I think we should go ahead and merge this. The warning box could optionally be added in a follow-up PR.

**Pull request merged**